sos/policies/runtimes/crio.py

```diff
@@ -47,13 +49,21 @@ def get_images(self):
         """
         images = []
         if self.active:
-            out = sos_get_command_output("%s images" % self.binary,
+            out = sos_get_command_output("%s images -o json" % self.binary,
                                          chroot=self.policy.sysroot)
             if out['status'] == 0:
-                for ent in out['output'].splitlines():
-                    ent = ent.split()
-                    # takes the form (image_name, image_id)
-                    images.append((ent[0] + ':' + ent[1], ent[2]))
+                out_json = json.loads(out["output"])
+                for image in out_json["images"]:
+                    # takes the form (repository:tag, image_id)
+                    if len(image["repoTags"]) > 0:
+                        for repo_tag in image["repoTags"]:
+                            images.append((repo_tag, image["id"]))
+                    else:
+                        if len(image["repoDigests"]) == 0:
+                            image_name = "<none>"
+                        else:
+                            image_name = image["repoDigests"][0].split("@")[0]
+                        images.append((image_name + ":<none>", image["id"]))
```

Comment:
I don't think adding `:<none>` here as part of the image name is the right move. That would require plugins to either use that specifically (self.add_cmd_output('foobar', container=`mycontainer:<none>`), or make their relevant container calls all use globs (`self.add_cmd_output('foobar', container=`mycontainer*`) which will probably add to problems down the line.

Reply:
so here I was trying to reproduce the output of table format for `crictl images` (also to be compatible with docker output), so adding none is what `crictl` does https://github.com/kubernetes-sigs/cri-tools/blob/ccb94be8f9c365f243d0c5c65fa7c26993d6094c/cmd/crictl/image.go#L522, image name can also be none https://github.com/kubernetes-sigs/cri-tools/blob/ccb94be8f9c365f243d0c5c65fa7c26993d6094c/cmd/crictl/image.go#L542
docker for command https://github.com/sosreport/sos/blob/main/sos/policies/runtimes/__init__.py#L122-L123 will return e.g.

```
# docker images --format '{{lower .Repository}}:{{lower .Tag}} {{lower .ID}}'
<none>:<none> dbfda3a8826c
quay.io/npinaeva/ovn-k:<none> a5851f835871

```
and for `self.add_cmd_output` we need container name and not image name - actually I don't think images are used in sos right now, just added that for completeness

Reply:
Yep, I went cross eyed there and was mixing images with IDs.